types/wlr_box.c

```diff
@@ -11,17 +11,17 @@ void wlr_box_closest_point(const struct wlr_box *box, double x, double y,
 	// find the closest x point
 	if (x < box->x) {
 		*dest_x = box->x;
-	} else if (x > box->x + box->width) {
-		*dest_x = box->x + box->width;
+	} else if (x >= box->x + box->width) {
```

Comment:
Hmm, why this change?

Reply:
I made this change a few months back, so a bit hard for me to remember, but IIRC it caused an issue where some sides of the confine region would not confine, since the closest point wouldn't be inside the box anymore, so the constraint would be deactivated.

Reply:
This change doesn't make sense in this function. Try reverting it and understand why it breaks stuff.